### DISCUSSION TITLE
Where to get my github commit SHA for sentry setup

### DISCUSSION THREAD

**Discussion by Pattygeek** [vercel/next.js#15849] [Help]
I am setting up my project for sentry, I need to get my github commit SHA, from what i discovered each commit has it's own SHA, b ut what the setup is requesting seems to me like a single value. Please, how do i go about this?

**Comment by jamesmosier** [comment]
If you are hosting with Vercel they provide some environment variables you could use to get the latest Git SHA: https://github.com/vercel/next.js/blob/668cc6a6f1901f0c0752ef45ab730d52550e8a0e/examples/with-sentry/next.config.js#L18-L21

Otherwise you'd need to get it yourself (this would go in your `next.config.js` near where you initialize the Sentry plugin):

```js
const cp = require('child_process');
const gitSha = cp.execSync('git rev-parse HEAD', {
  cwd: __dirname,
  encoding: 'utf8'
});
```

or the short version of the SHA:

```js
const cp = require('child_process');
const gitSha = cp.execSync('git rev-parse --short HEAD', {
  cwd: __dirname,
  encoding: 'utf8'
});
```
